### DISCUSSION TITLE
Can I run on a git URL?

### DISCUSSION THREAD

**Discussion by detj** [o2sh/onefetch#1357] [Q&A] [answered]
As I understand, onefetch doesn't clone or fetch from https/ssh git URLs. Can you elaborate on the reasons behind such a stance?

If I want to run onefetch on an existing GitHub repo and all I have is the URL, what is the best or fastest way I can? (Assuming I don't want to manually clone the repo.)

<img width="1094" alt="image" src="https://github.com/o2sh/onefetch/assets/280236/f1a6a805-c349-4231-b0f6-c5808e50f22e">

**Comment by spenserblack** [comment]
I think part of the reason is that it's really trivial to clone a repository to a temporary directory, and can be written with a 3-line script.

In general we've also been able to keep this an offline-only tool, and there are some drawbacks to supporting actions that require a connection to the remote (more dependencies, more usage to support, etc.)

**Comment by ccoVeille** [comment, marked as answer]
> Assuming I don't want to manually clone the repo.

If you don't then the tool will have to. It's not in the tool scope. It would add complexity, then will come feature such as: - "please respect http proxy when dealing with http URL"
- "please retry downloading my internet connection sucks"
- please add Gitlab supports
- private repository

And then you implemented a whole factory instead of a DRY tool.

I agree cloning first via a script.

My point would have been different if the idea was I don't know parsing something and calling a few random URL discovered.

But here, you somehow explain you:
- have internet
- you have the repository URL
- you expect to pass the URL to onefetch

**Comment by detj** [comment]
Thanks for the replies. Sure, like I said, as users we can all clone our git repos. I just wanted to understand the scope and the principles better. I agree, it would open up a can of worms like http proxies, supporting providers.
